File: 프로그래머스/lv2/42842. 카펫/카펫.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    var w = 0, h = 0
    let sum = brown + yellow
    for i in 1...sum {
        if sum%i == 0 { // 나누어 떨어지는 수 중에
            w = sum/i   // 가로 세로를 구한다음
            h = i       // 공식이 해당되는지 확인
            if (w-2) * (h-2) == yellow {
                break
            }
        }
    }
    return [w, h]
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    _sum = brown + yellow
    for i in range(3, brown+1):   # 세로
        for j in range(3, brown+1):    # 가로
            if i*j == _sum and (i-2)*(j-2) == yellow:
                return [j, i]